java/core/src/java/org/apache/orc/impl/ParserUtils.java

```diff
@@ -244,80 +250,172 @@ public static TypeDescription parseType(ParserUtils.StringPosition source) {
 
   private static final Pattern INTEGER_PATTERN = Pattern.compile("^[0-9]+$");
 
+  public static ColumnVector[] findBranchVectors(TypeDescription schema,
+                                                 ParserUtils.StringPosition source,
+                                                 VectorizedRowBatch batch) {
+    List<String> names = ParserUtils.splitName(source);
+    return vectorWalker.get().find(schema, names, true, batch).vectorBranch;
+  }
+
   public static TypeDescription findSubtype(TypeDescription schema,
                                             ParserUtils.StringPosition source) {
     return findSubtype(schema, source, true);
   }
 
-  public static TypeDescription findSubtype(TypeDescription schema,
-                                            ParserUtils.StringPosition source,
-                                            boolean isSchemaEvolutionCaseAware) {
-    List<String> names = ParserUtils.splitName(source);
-    if (names.size() == 1 && INTEGER_PATTERN.matcher(names.get(0)).matches()) {
-      return schema.findSubtype(Integer.parseInt(names.get(0)));
+  private static final ThreadLocal<TypeWalker> typeWalker = ThreadLocal.withInitial(TypeWalker::new);
```

Comment:
I don't think we need the ThreadLocals. This code is only called at setup.

Reply:
Make sense, removed this. Given that ColumnVectors are cached, this should not happen after the initial setup.